source/game_sa/Entity/Ped/CCopPed.cpp

```diff
@@ -1 +1,247 @@
+        SetWeaponAccuracy(60);
+        break;
+    }
+    }
+    m_pIntelligence->SetDmRadius(60.0f);
+    m_pIntelligence->SetNumPedsToScan(8);
+    m_pedSpeech.Initialise(this);
+
+    /* rest done by the compiler, set as default value */
+}
+
+// 0x5DDC60
+CCopPed* CCopPed::Constructor(uint32_t copTypeOrModelID) {
+    this->CCopPed::CCopPed(copTypeOrModelID);
+    return this;
+}
+
+// 0x5DE0D0
+CCopPed::~CCopPed() {
+    if (FindPlayerPed())
+        FindPlayerWanted()->RemovePursuitCop(this);
+    SetPartner(nullptr);
+    ClearCriminalsToKill();
+}
+
+// 0x5DE0D0
+CCopPed* CCopPed::Destructor() {
+    this->CCopPed::~CCopPed();
+    return this;
+}
+
+void Test(CEntity** p) {
+
+}
+
+// 0x5DDE80
+void CCopPed::SetPartner(CCopPed* pPartner) {
+    if (m_pCopPartner)
+        m_pCopPartner->CleanUpOldReference(reinterpret_cast<CEntity**>(&m_pCopPartner));
+    m_pCopPartner = pPartner;
+    if (pPartner)
+        pPartner->RegisterReference(reinterpret_cast<CEntity**>(&m_pCopPartner));
+}   
+
+// NOTSA
+void CCopPed::ClearCriminalListFromDeadPeds() {
+    for (CPed*& v : m_apCriminalsToKill) {
+        if (v && v->m_fHealth <= 0.0f) {
+            v->CleanUpOldReference(reinterpret_cast<CEntity**>(&v));
+            v = nullptr;
+        }
+    }
+}
+
+// NOTSA
+bool CCopPed::IsCriminalInList(CPed* pCriminal) {
+    assert(pCriminal); // NOTSA
+    for (CPed* v : m_apCriminalsToKill) {
+        if (v == pCriminal)
+            return true;
+    }
+    return false;
+}
+
+// NOTSA
+size_t CCopPed::GetEmptySlotForCriminalToKill() {     
+    for (size_t i = 0; i < std::size(m_apCriminalsToKill); i++) {
+        if (!m_apCriminalsToKill[i])
+            return i;
+    }
+    return -1;
+}
+
+// 0x5DDEB0
+void CCopPed::AddCriminalToKill(CPed* pCriminal) {
+    if (!pCriminal)
+        return;
+    if (pCriminal->IsPlayer())
+        return;
+    switch (pCriminal->m_nPedType) {
+    case ePedType::PED_TYPE_COP:
+    case ePedType::PED_TYPE_MEDIC:
+    case ePedType::PED_TYPE_FIREMAN:
+        return;
+    }
+    if ((unsigned)pCriminal->m_nPedType >= (unsigned)ePedType::PED_TYPE_MISSION1)
+        return;
+    if (pCriminal->IsCreatedBy(ePedCreatedBy::PED_MISSION))
+        return;
+    ClearCriminalListFromDeadPeds();
+    if (pCriminal->m_fHealth <= 0.0f)
+        return;
+    if (IsCriminalInList(pCriminal))
+        return;
+    /* Find slot, put criminal in there */
+    {
+        size_t slot = GetEmptySlotForCriminalToKill();
+        if (slot == -1)
+            return; /* no free slot, return */
+        pCriminal->RegisterReference(reinterpret_cast<CEntity**>(&m_apCriminalsToKill[slot]));
+        m_apCriminalsToKill[slot] = pCriminal;
+    }
+
+    pCriminal->bWantedByPolice = true;
+    pCriminal->bCullExtraFarAway = true;
+    pCriminal->m_fRemovalDistMultiplier = 0.3f;
+    pCriminal->m_nTimeTillWeNeedThisPed = CTimer::m_snTimeInMilliseconds + 300'000;
```

Comment:
```suggestion
    pCriminal->m_nTimeTillWeNeedThisPed = CTimer::m_snTimeInMilliseconds + 300'000;
```
Hmm, why do I see `'` in `300'000` ?

Reply:
Why not?

Reply:
Because It's C++14 feature...

Reply:
Well that's sad.

Reply:
```suggestion
    pCriminal->m_nTimeTillWeNeedThisPed = CTimer::m_snTimeInMilliseconds + 300000; // 300k
```